application/src/main/java/org/togetherjava/tjbot/commands/CommandHandler.java

```diff
@@ -0,0 +1,75 @@
+package org.togetherjava.tjbot.commands;
+
+import net.dv8tion.jda.api.events.interaction.ButtonClickEvent;
+import net.dv8tion.jda.api.events.interaction.SelectionMenuEvent;
+import net.dv8tion.jda.api.events.interaction.SlashCommandEvent;
+import net.dv8tion.jda.api.hooks.ListenerAdapter;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * The command handler
+ *
+ * Commands need to be added to the commandList
+ */
+public class CommandHandler extends ListenerAdapter {
+    private final static Logger logger = LoggerFactory.getLogger(CommandHandler.class);
+
+    private final List<Command> commandList = new ArrayList<>();
+    private final Map<String, Command> commandMap;
+
+    public CommandHandler() {
+        commandList.addAll(List.of(new ReloadCommand(this)
+        // add your command here
+        ));
+
+        commandMap = commandList.stream()
+            .collect(Collectors.toMap(Command::getCommandName, command -> command));
+    }
+
+    @Override
+    public void onSlashCommand(@NotNull SlashCommandEvent event) {
+        Command command = commandMap.get(event.getName());
+
+        if (command != null) {
+            command.onSlashCommand(event);
+        } else {
+            logger.error("Command '{}' doesn't exist?", event.getName());
+        }
+    }
+
+    @Override
```

Comment:
You're repeating yourself [here](https://github.com/Together-Java/TJ-Bot/pull/37/files#diff-e5f33a13deeba755c63c3da6a06821b5b943b870e8a92c7b05ed588da51ba6dbR48-R70), I'd say extract this to a separate method.